src/ORM/DataObject.php

```diff
@@ -3294,7 +3295,7 @@ public function getReverseAssociation($className)
      * @param string|array $limit A limit expression to be inserted into the LIMIT clause.
      * @param string $containerClass The container class to return the results in.
      *
-     * @return DataList The objects matching the filter, in the class specified by $containerClass
+     * @return DataList<static> The objects matching the filter, in the class specified by $containerClass
```

Comment:
Yes! this is the change we want to see ❤️

Reply:
Yeah it's technically wrong when you go `MyClass::get(DifferentClass::class)` (returns `DataList<MyClass>`) and imprecise when using `DataObject::get(DifferentClass::class)` (returns `DataList<DataObject>`. But as I mentioned in the issue I think those are uncommon enough that it will be worthwhile - and the latter case is a bad smell anyway.